web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Profile/StoreInfo/page/UpdateStoreInfoForm/index.tsx

```diff
@@ -0,0 +1,228 @@
+      }
+    },
+    [app?.id, updateAppInfoMutation, teamId, appId],
+  );
+
+  return (
+    <form className="grid gap-y-7" onSubmit={handleSubmit(submit)}>
+      <Typography variant={TYPOGRAPHY.H7}>Permissions</Typography>
+      <div className="grid grid-cols-auto/1fr py-6 px-5 border-[1px] rounded-xl border-grey-200 gap-x-4">
+        <Checkbox
+          register={register("is_developer_allow_listing")}
+          disabled={!isEditable || !isEnoughPermissions}
+        />
+        <div className="grid gap-y-2">
+          <Typography variant={TYPOGRAPHY.R3} className="text-grey-700">
+            Allow App Store listing
+          </Typography>
+          <Typography variant={TYPOGRAPHY.R4} className="text-grey-400">
+            Once you submit your app for review, it can be placed in Worldcoin
+            App Store, if it’s chosen to be displayed by the Worldcoin team.
+          </Typography>
+        </div>
+      </div>
+      <div className="grid gap-y-3">
+        <Typography variant={TYPOGRAPHY.H7} className="text-grey-900">
+          App description
+        </Typography>
+        <Typography variant={TYPOGRAPHY.R3} className="text-grey-500">
+          Describe your app integration to possible users
+        </Typography>
+      </div>
+      <div className="grid gap-y-5">
+        <TextArea
+          label="Overview"
+          required
+          rows={5}
+          errors={errors.description_overview}
+          disabled={!isEditable || !isEnoughPermissions}
+          placeholder="Describe the project for the users who would like to try your integration"
+          register={register("description_overview")}
+        />
+        <TextArea
+          label="How it works"
+          rows={5}
+          errors={errors.description_how_it_works}
+          disabled={!isEditable || !isEnoughPermissions}
+          placeholder="How do users interact with World ID in your app?"
+          register={register("description_how_it_works")}
+        />
+        <TextArea
+          label="How to connect"
+          rows={5}
+          errors={errors.description_connect}
+          disabled={!isEditable || !isEnoughPermissions}
+          placeholder="Explain, if required, how users should set up this app to start using World ID."
+          register={register("description_connect")}
+        />
+        <Input
+          label="World App Description"
+          maxLength={50}
+          errors={errors.world_app_description}
+          disabled={!isEditable || !isEnoughPermissions}
+          placeholder="Short description for display in the app"
+          register={register("world_app_description")}
+          addOnRight={
+            <Typography variant={TYPOGRAPHY.R5} className="text-grey-400">
+              {remainingCharacters}
+            </Typography>
+          }
+        />
+      </div>
+      <DecoratedButton
+        type="submit"
+        className="w-40 h-12"
+        disabled={!isEditable || !isEnoughPermissions}
```

Comment:
@andy-t-wang You can get also `isValid` from `formState`. And maybe it's a good idea to check `dirtyFields` from `formState` too, to block button if there is no changes in the form. What do you think?